plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -4017,4 +4018,25 @@ function wc_update_product_archive_title( $post_type_name, $post_type ) {
 }
 add_filter( 'post_type_archive_title', 'wc_update_product_archive_title', 10, 2 );
 
+/**
+ * Checks the old and current themes and determines if the "wc_blocks_use_blockified_product_grid_block_as_template"
+ * option need to be updated accordingly.
+ *
+ * @param string    $old_name Old theme name.
+ * @param \WP_Theme $old_theme Instance of the old theme.
+ * @return void
+ */
+function wc_check_should_use_blockified_product_grid_templates( $old_name, $old_theme ) {
```

Comment:
Should we not read the option first? Then we only need to update it if its different and is less expensive than writing so a slight optimization.

Reply:
Sounds good! I like making things less expensive. 😄

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/48382/commits/d9536dc4f8cc20dfe832e4b356a7372e47f1cc19